### PR TITLE
ENH: added instruments

### PR DESCRIPTION
**Pull request by sglyon**
Added facilities to automatically do the row-wise kronecker  product if a matrix of instruments is passed.

### CODE REVIEW

**Comment by sglyon** [comment]
Also adds tests for the `row_kron` function as well as `max_args`.

Fixed bug so default solver is applied to both `gmm` methods.

File: test/runtests.jl [hunk 1]
```diff
@@ -39,4 +39,38 @@ facts("Testing basic interface") do
...
+        a = randn(400, 3)
+        b = randn(400, 5)
+        out = GMM.row_kron(a, b)
+        @fact size(out) => (400, 15)
+
+        rows_good = true
+        for row=1:400
+            rows_good = out[row, :] == kron(a[row, :], b[row, :])
+        end
+        @fact rows_good => true
```
Comment (gragusa) [inline test/runtests.jl:60]: I think this should go inside the loop. Shouldn't it?

Comment (sglyon) [inline test/runtests.jl:60]: I didn't want to generate 400 tests for this simple test case. So what I mean to do was `rows_good &= out...` I'll fix that.

**Comment by gragusa** [comment]
Except for this [small issue](https://github.com/gragusa/GMM.jl/pull/21/files#diff-fce720c43af3c52c862fd7451c7374b8R60) the pull request seems good to go.

**Comment by sglyon** [comment]
I think the latest commit fixed that problem. Let me know if not

**Comment by sglyon** [comment]
I think will end up having the same conversation as in #14 about closing over the instrument data. When we decide how to handle that data, we should apply the same to this

**Comment by gragusa** [comment]
Yes, that works too.

**Comment by gragusa** [comment]
Can you add this

```
[![Coverage Status](https://coveralls.io/repos/gragusa/GMM.jl/badge.svg?branch=master)](https://coveralls.io/r/gragusa/GMM.jl?branch=master)
```

to the README so that coverall badge will appear? Then we can merge.

**Pull request merged**